### PR TITLE
Jm/v2

### PR DESCRIPTION
**Pull request by jawwadmukhtargmail**

### CODE REVIEW

File: CommunicationService/CommunicationService.AzureFunction/ProcessJobQueue.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            _messageFactory = messageFactory;
+        }
+
+        [FunctionName("ProcessJobQueue")]
+        public void Run([ServiceBusTrigger("job", Connection = "AzureWebJobsStorage")]string myQueueItem, ILogger log)
+        {
+
+            SendCommunicationRequest sendCommunicationRequest = JsonConvert.DeserializeObject<SendCommunicationRequest>(myQueueItem);
+            IMessage message = _messageFactory.Create(sendCommunicationRequest);
+            List<int> recipients = message.IdentifyRecipients(sendCommunicationRequest.RecipientUserID, sendCommunicationRequest.JobID, sendCommunicationRequest.GroupID);
```
Comment (mike-f50) [inline CommunicationService/CommunicationService.AzureFunction/ProcessJobQueue.cs:30]: I think one job could result in different messages to different people.  Maybe IdentifyRecipients should return Dictionary<int, EmailTemplate>

File: CommunicationService/CommunicationService.AzureFunction/ProcessRecipientQueue.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            _messageFactory = messageFactory;
+            _sendEmailService = sendEmailService;
+        }
+
+        [FunctionName("ProcessRecipientQueue")]
+        public void Run([ServiceBusTrigger("recipient", Connection = "AzureWebJobsStorage")]string myQueueItem, ILogger log)
+        {
+            SendCommunicationRequest sendCommunicationRequest = JsonConvert.DeserializeObject<SendCommunicationRequest>(myQueueItem);
+            IMessage message = _messageFactory.Create(sendCommunicationRequest);
+            SendGridData sendGridData = message.PrepareTemplateData(sendCommunicationRequest.RecipientUserID, sendCommunicationRequest.JobID, sendCommunicationRequest.GroupID).Result;
```
Comment (mike-f50) [inline CommunicationService/CommunicationService.AzureFunction/ProcessRecipientQueue.cs:31]: Do we need to pass RecipientUserID etc. in here, or can message get these via the Create method called above?

File: CommunicationService/CommunicationService.Core/Domains/SendGridData.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using CommunicationService.Core.Domains;
+
+namespace CommunicationService.Core.Domains
+{
+    public class SendGridData
```
Comment (mike-f50) [inline CommunicationService/CommunicationService.Core/Domains/SendGridData.cs:5]: Should we use a more generic class name, e.g. EmailBuildData?

File: CommunicationService/CommunicationService.MessageService/MessageFactory.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+        public MessageFactory(IConnectUserService connectUserService, IConnectRequestService connectRequestService, IQueueClient queueClient)
+        {
+            _connectUserService = connectUserService;
+            _connectRequestService = connectRequestService;
+            _queueClient = queueClient;
+        }
+        public IMessage Create(SendCommunicationRequest sendCommunicationRequest)
+        {
+            switch (sendCommunicationRequest.EmailTemplate.EmailTypes)
```
Comment (mike-f50) [inline CommunicationService/CommunicationService.MessageService/MessageFactory.cs:28]: Should EmailType be a direct child of sendCommunicationRequest?  I think EmailTemplate comes later - and there might be several per EmailType.

File: CommunicationService/CommunicationService.AzureFunction/ProcessRecipientQueue.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+using CommunicationService.Core.Interfaces.Services;
+using CommunicationService.MessageService;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace CommunicationService.AzureFunction
+{
+    public class ProcessRecipientQueue
```
[the remark this answers is not in the document]
Comment (jawwadmukhtargmail) [reply CommunicationService/CommunicationService.AzureFunction/ProcessRecipientQueue.cs:15]: done

File: CommunicationService/CommunicationService.Core/Domains/Entities/Request/EmailTypes.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace CommunicationService.Core.Domains.Entities.Request
+{
+    public enum EmailTypes
```
[the remark this answers is not in the document]
Comment (jawwadmukhtargmail) [reply CommunicationService/CommunicationService.Core/Domains/Entities/Request/EmailTypes.cs:7]: done

File: CommunicationService/CommunicationService.Core/Domains/Entities/Request/EmailTypes.cs [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    {
+        Welcome = 1,
+        Chasers = 2,
+        TaskNotification = 3,
+        TaskDailyDigest = 4,
+        TaskRequesterUpdate = 5
+    }
```
[the remark this answers is not in the document]
Comment (jawwadmukhtargmail) [reply CommunicationService/CommunicationService.Core/Domains/Entities/Request/EmailTypes.cs:14]: done

File: CommunicationService/CommunicationService.AzureFunction/ProcessJobQueue.cs (hunk 1 above)
Comment (jawwadmukhtargmail) [reply CommunicationService/CommunicationService.AzureFunction/ProcessJobQueue.cs:30]: done

**Pull request merged**